legend-engine-xts-persistence/legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/ingestmode/UnitemporalSnapshotAbstract.java

```diff
@@ -89,6 +98,29 @@ default void validate()
                     throw new IllegalStateException(String.format("Can not build UnitemporalSnapshot, partitionKey: [%s] not specified in partitionFields", partitionKey));
                 }
             }
+            int partitionKeysWithMoreThanOneValues = 0;
+            for (Set<String> partitionValues: partitionValuesByField().values())
+            {
+                if (partitionValues.size() > 1)
+                {
+                    partitionKeysWithMoreThanOneValues++;
+                }
+            }
+            if (partitionKeysWithMoreThanOneValues > 1)
+            {
+                throw new IllegalStateException(String.format("Can not build UnitemporalSnapshot, partitionValuesByField does not support more than 1 value for more than one partition key"));
+            }
+        }
+
+        if (!partitionSpecList().isEmpty())
+        {
+            for (Map<String, String> partitionSpec : partitionSpecList())
+            {
+                if (partitionFields().size() != partitionSpec.size())
+                {
+                    throw new IllegalStateException("Can not build UnitemporalSnapshot, size of each partitionSpec must be same as size of partitionFields");
+                }
+            }
         }
```

Comment:
So we are still expecting the users to pass both partitionFields and partitionSpecList in this use case - so shall we validate the the partition keys in the spec list match the ones specified in partitionFields? (we are doing this for the partitionValuesByField case, so just  wondering if we should do this here too for consistency)

Reply:
Good point, will add it.